File: src/leetcode/climbing-stairs.py
## https://leetcode.com/problems/climbing-stairs/

class Solution:
        
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        if n == 2:
            return 2
        
        step1 = 2
        step2 = 1

        total_steps = 0

        for _ in range(3, n+1):
            total_steps = step1 + step2
            step2 = step1
            step1 = total_steps

        return total_steps

solution = Solution()

solution.climbStairs(38)

File: src/google/ClimbingStairs.java
// https://leetcode.com/problems/climbing-stairs/description/

public class ClimbingStairs {
    private int pathsTaken = 0;
    public int climbStairs(int n) {
        int pathCounter[] = new int[n+1];
        pathCounter[1] = 1;
        pathCounter[2] = 2;
        
        for(int step = 3; step <= n; step++) {
            pathCounter[step] = pathCounter[step-2] * 2 + 1;
        }

        return pathCounter[n];
    }
}